build: Drop make dist in gitian builds

hebasto opened the pull request:
After the merge of #18331, the packaged source tarball is created by `git archive`, but the binaries are built from another one which is made by `make dist`.

With this PR the only source tarball, created by `git archive`, is used both for binaries building and for packaging to users.

Close #16588.
Close #18547.

As a good side-effect, #18349 becomes redundant.

**Change in behavior**

The following variables https://github.com/bitcoin/bitcoin/blob/1b151e3ffce7c1a2ee46bf280cc1d96775d1f91e/configure.ac#L2-L6

are no longer used for naming of directories and tarballs.

Instead of them the gitian descriptors use a git tag (if available) or a commit hash.

---

Also a small refactor commit picked from #18404.

hebasto committed:
build: Drop make dist in gitian builds

hebasto committed:
refactor: Drop unused ${WRAP_DIR}/${HOST} directory

This commit removes the directory that is no longer used since #16667.

MarcoFalke commented:
Open-Close to re-run ci. See https://github.com/bitcoin/bitcoin/issues/15847#issuecomment-487591130

MarcoFalke commented:
Travis run here: https://travis-ci.org/github/bitcoin/bitcoin/builds/672303495 . Not sure why it is not linked.

laanwj commented:
Concept ACK

laanwj commented:
Concept ACK

MarcoFalke commented:
Looks like `bitcoin-0.19.99-win64-setup-unsigned.exe` is named incorrectly

hebasto commented:
Updated 8d0c980e6425012f772c89ff4b8346ba3f459151  -> 19a51a83454b39698b76651a40e5ebf31c61b234 ([pr18556.01](https://github.com/hebasto/bitcoin/commits/pr18556.01) -> [pr18556.02](https://github.com/hebasto/bitcoin/commits/pr18556.02), [diff](https://github.com/hebasto/bitcoin/compare/pr18556.01..pr18556.02)):

- addressed @MarcoFalke's [comment](https://github.com/bitcoin/bitcoin/pull/18556#issuecomment-611689873)
> Looks like `bitcoin-0.19.99-win64-setup-unsigned.exe` is named incorrectly

MarcoFalke commented on contrib/gitian-descriptors/make_git_archive:
```diff
@@ -0,0 +1,20 @@
+  if [[ $RECENT_TAG == v* ]]; then
+    VERSION=${RECENT_TAG:1}
+  else
+    VERSION=$RECENT_TAG
```
what is this condition for?

hebasto replied:
We tag releases with the prepended "v", like "v0.20.0rc1".

This `if` statement just removes it on purpose to include the version string (e.g., "0.20.0rc1") into file names.

MarcoFalke replied:
Oh, I meant the else branch

hebasto replied:
```
if [ $RECENT_TAG = $(git describe HEAD) ]; then
  if [[ $RECENT_TAG == v* ]]; then
    VERSION=${RECENT_TAG:1} # Set VERSION to release (with "v") tagged commit
  else
    VERSION=$RECENT_TAG # Set VERSION to non-release (without "v") tagged commit
  fi
else
  VERSION=$(git rev-parse --short HEAD) # Set VERSION to untagged commit
fi
DISTNAME=bitcoin-${VERSION} # Use VERSION in file names
```

MarcoFalke replied:
Why would there be a non-v* tagged commit? This should probably be an assert(false)?

hebasto replied:
It is convenient for making gitian test builds. At least, for me :)

MarcoFalke replied:
ok, will leave this for other reviewers to bikeshed :)

fanquake replied:
> It is convenient for making gitian test builds. At least, for me :)

I mentioned on IRC, however I'll just reiterate here, I'd definitely normally NACK unclear/undocumented additions to our build system / release scripts. However given that this should be removed in a follow up PR I think it's ok as is. If you've got patches that are helpful for testing, you can maintain them locally, but please don't just bundle them into a PR along with other changes.

hebasto replied:
**The rationale for adding that addition:**
Currently, the `gitian-build.py` script allows to make builds from personal repos by passing the `--url` command line argument. Also a branch could be chosen. This change adds an option to point the script to a tagged commit on personal repo as well.

MarcoFalke commented:
ACK, can squash the first two commits?

hebasto commented:
Updated 19a51a83454b39698b76651a40e5ebf31c61b234  -> 2aa48edec0101f8a77a2189244fc62722ff7a123 ([pr18556.02](https://github.com/hebasto/bitcoin/commits/pr18556.02) -> [pr18556.03](https://github.com/hebasto/bitcoin/commits/pr18556.03), [diff](https://github.com/hebasto/bitcoin/compare/pr18556.02..pr18556.03)):

- addressed @MarcoFalke's [comment](https://github.com/bitcoin/bitcoin/pull/18556#issuecomment-615228781)
> ACK, can squash the first two commits?

MarcoFalke commented:
ACK 2aa48edec0101f8a77a2189244fc62722ff7a123

dongcarl commented:
Will take a look this week... Testing a few patches on top of this branch RN.

dongcarl commented:
@hebasto I've opened #18741 based on this! I think that perhaps ab65fe2555eb59633578b4f89c525470e36dd1c3 _could_ be pulled into this PR, but it kind of doesn't make sense without the Guix changes, so I'll leave that decision up to you :relaxed:

dongcarl commented:
ACK 2aa48edec0101f8a77a2189244fc62722ff7a123
Don't want to delay merging this so let's just do the Guix parts afterwards in #18741

adamjonas commented:
This should close #16588 if I'm not mistaken.

hebasto commented:
> This should close #16588 if I'm not mistaken.

The OP has been updated.

MarcoFalke commented:
@fanquake Anything left to do here?

fanquake commented on test/lint/lint-shell.sh:
```diff
@@ -46,15 +46,17 @@ if ! command -v yq > /dev/null; then
+SHELLCHECK_CMD="shellcheck --external-sources --check-sourced $EXCLUDE_GITIAN"
 for descriptor in $(git ls-files -- 'contrib/gitian-descriptors/*.yml')
 do
-    echo
```
I don't know is this is expected or changed here, but running this I see:
```bash
test/lint/lint-shell.sh
2020/04/25 19:36:23 unknown command "contrib/gitian-descriptors/gitian-linux.yml" for "yq"
2020/04/25 19:36:23 unknown command "contrib/gitian-descriptors/gitian-osx-signer.yml" for "yq"
2020/04/25 19:36:23 unknown command "contrib/gitian-descriptors/gitian-osx.yml" for "yq"
2020/04/25 19:36:23 unknown command "contrib/gitian-descriptors/gitian-win-signer.yml" for "yq"
2020/04/25 19:36:23 unknown command "contrib/gitian-descriptors/gitian-win.yml" for "yq"
```

hebasto replied:
Could #17361 help? Did you have [`yq`](https://kislyuk.github.io/yq/) installed?

fanquake replied:
> Did you have yq installed?

Yes. `yq version 3.3.0`

hebasto replied:
> > Did you have yq installed?
> 
> Yes. `yq version 3.3.0`

Not [that](https://github.com/mikefarah/yq/releases) but [this](https://kislyuk.github.io/yq/) one:
```
$ brew install python-yq
```

fanquake reviewed (approved):
ACK 2aa48edec0101f8a77a2189244fc62722ff7a123 - I've had a quick look over this, and don't want to block merging if this actually gets as closer to finally having this all sorted out. Obviously we've still got #18741, and after speaking to Carl this morning, there will likely be even more changes after that (not Guix specific).
